src/js/components/history/HistoryItem.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {View, parse, Spec} from 'vega';
+import {HistoryRecord, HistoryState} from '../../store/factory/History';
+import {cleanSpecForPreview} from '../../ctrl/demonstrations';
+import {startDragging, stopDragging} from '../../actions/inspectorActions';
+import {DraggingStateRecord, HistoryDraggingState} from '../../store/factory/Inspector';
+import sg from '../../ctrl/signals';
+import {MODE, SELECTED, CELL} from '../../store/factory/Signal';
+import {NumericValueRef, StringValueRef, tupleid} from 'vega';
+import {channelName} from '../../actions/bindChannel';
+import {setMarkVisual} from '../../actions/markActions';
+import * as vega from 'vega';
+import bindChannel from '../../actions/bindChannel';
+import {ColumnRecord, Schema} from '../../store/factory/Dataset';
+import { AnyAction } from 'redux';
+import {ThunkDispatch} from 'redux-thunk';
+import {State} from '../../store';
+import {setSignal} from '../../actions/signalActions';
+import {SignalValue} from 'vega-typings/types';
+import {batchGroupBy} from '../../reducers/historyOptions';
+
+const ctrl = require('../../ctrl');
+
+
+interface OwnProps {
+  id: string,
+  history: HistoryRecord // TODO: use History.ts for you just have to pass the id
+
+  groupNames: any[];
+}
+interface DispatchProps {
+  startDragging: (d: DraggingStateRecord) => void;
+  stopDragging: () => void;
+  setMarkVisual: (payload: {property: string, def: NumericValueRef | StringValueRef}, markId: number) => void;
+
+  bindChannel: (dsId: number, field: ColumnRecord, markId: number, property: string) => void;
+
+  setSignal: (value: SignalValue, signal: string) => void;
+}
+
+function mapDispatchToProps(dispatch: ThunkDispatch<State, null, AnyAction>, ownProps: OwnProps): DispatchProps {
+  return {
+    startDragging: (d: DraggingStateRecord) => {
+      dispatch(startDragging(d)); },
+    stopDragging: () => {
+      dispatch(stopDragging()); },
+    setMarkVisual: (p, markId) => {
+      dispatch(setMarkVisual(p, markId));
+    },
+    bindChannel: (dsId: number, field: ColumnRecord, markId: number, property: string) => {
+      dispatch(bindChannel(dsId, field, markId, property));
+    },
+    setSignal: (value: SignalValue, signal: string) => {
+      dispatch(setSignal(value, signal));
+    }
+  };
+}
```

Comment:
(style)
Since your dispatch functions have no side effects, you could also use the `actionCreators` syntax here. See `Interaction.tsx` line 139 for an example.